core/src/Spirit/Chain.cpp

```diff
@@ -9,710 +9,726 @@
 {
-	int idx_image = -1;
-	std::shared_ptr<Data::Spin_System> image;
-	std::shared_ptr<Data::Spin_System_Chain> chain;
+    int idx_image = -1;
     
-    // Fetch correct indices and pointers
     try
     {
+    	std::shared_ptr<Data::Spin_System> image;
+    	std::shared_ptr<Data::Spin_System_Chain> chain;
+        
+        // Fetch correct indices and pointers
         from_indices( state, idx_image, idx_chain, image, chain );
+
+    	for (unsigned int i = 0; i < chain->E_interpolated.size(); ++i)
+    	{
+    		E_interpolated[i] = (float)chain->E_interpolated[i];
+    	}
     }
-    catch( const Utility::Exception & ex )
+    catch( ... )
     {
-        Utility::Handle_exception( ex, idx_image, idx_chain );
-        return ;
+        Utility::Handle_Exception( idx_image, idx_chain );
     }
-
-	for (unsigned int i = 0; i < chain->E_interpolated.size(); ++i)
-	{
-		E_interpolated[i] = (float)chain->E_interpolated[i];
-	}
 }
 
-std::vector<std::vector<float>> Chain_Get_Energy_Array_Interpolated(State * state, int idx_chain_i)
+std::vector<std::vector<float>> Chain_Get_Energy_Array_Interpolated( State * state, int idx_chain )
 {
-	int idx_image = -1, idx_chain = idx_chain_i;
-	std::shared_ptr<Data::Spin_System> image;
-	std::shared_ptr<Data::Spin_System_Chain> chain;
-    
+    int idx_image = -1;
     
-    std::vector<std::vector<float>> E_arr_interpolated ( chain->E_array_interpolated.size() );
-    for (unsigned int i=0; i < chain->E_array_interpolated.size(); i++ )
-        E_arr_interpolated[i] = std::vector<float> ( chain->E_array_interpolated[i].size(), 0 );
-    
-    // Fetch correct indices and pointers
     try
     {
+    	std::shared_ptr<Data::Spin_System> image;
+    	std::shared_ptr<Data::Spin_System_Chain> chain;
+        
+        std::vector<std::vector<float>> E_arr_interpolated ( chain->E_array_interpolated.size() );
+        for (unsigned int i=0; i < chain->E_array_interpolated.size(); i++ )
+            E_arr_interpolated[i] = std::vector<float> ( chain->E_array_interpolated[i].size(), 0 );
+        
+        // Fetch correct indices and pointers
         from_indices( state, idx_image, idx_chain, image, chain );
+        
+        for (unsigned int i=0; i < chain->E_array_interpolated.size(); i++)
+            for ( unsigned int j=0; j < chain->E_array_interpolated[i].size(); j++)
+                E_arr_interpolated[i][j] = (float) chain->E_array_interpolated[i][j];
+                
+    	return E_arr_interpolated;
     }
-    catch( const Utility::Exception & ex )
+    catch( ... )
     {
-        Utility::Handle_exception( ex, idx_image, idx_chain );
-        return E_arr_interpolated;
+        Utility::Handle_Exception( idx_image, idx_chain );
+        
+        // XXX: what should we return in that situation
+        std::vector<std::vector<float>> E_at_failure ( 1 );
+            E_at_failure[0] = std::vector<float> ( 1, 0 );
+        return E_at_failure;
```

Comment:
What you say is correct, so it makes sense what you wrote. It may result in an error in user code, if this is not expected, but that is not an issue for the API.